testing/tests/test_e2e/test_state.py

```diff
@@ -251,6 +253,17 @@ def pre_event(charm: CharmBase):
     }
 
 
+@pytest.mark.parametrize("id", ("", None, "28"))
+def test_checkinfo_changeid(id: Optional[str]):
+    info = CheckInfo("foo", change_id=ops.pebble.ChangeID(id))
+    if id is None:
+        assert info.change_id, "None should result in a random change_id"
+        info2 = CheckInfo("foo")
+        assert info.change_id != info2.change_id
+    else:
+        assert info.change_id == ops.pebble.ChangeID(id)
```

Comment:
If None is important, let's split this test into two, as there's almost no commonality vis a vis "" or "28.